proto/viam/app/v1/app.proto

```diff
@@ -1018,3 +1021,13 @@ message ListKeysRequest {
 message ListKeysResponse {
   repeated APIKeyWithAuthorizations api_keys = 1;
 }
+
+message RotateKeyRequest {
+  string key = 1;
+  string id =2;
+}
```

Comment:
so i think since we can remove `key` from this request since I think we decided you don't need the `key` to be able to do operations like rotate and delete. i do realize though that this will mess up the call to `IdentityAuthorizationGetForAPIKey` and [this call](https://github.com/viamrobotics/app/pull/2499/files#diff-5d4eeb9a95bb6dbd603dec16eb0fa3e6a5367a6bfa2e2379ee411c777afbfbb7R1951) until [APP-2776](https://viam.atlassian.net/browse/APP-2776) is done though. what i did to get around this for deleteKey is something like this: 
https://github.com/viamrobotics/app/blob/main/auth/permissioned_data.go#L548
https://github.com/viamrobotics/app/blob/main/data/data.go#L1948-L1951

Reply:
that makes sense. just pushed the change through